src/popover/popover-animation-from-top.jsx

```diff
@@ -58,56 +79,24 @@ const PopoverAnimationFromTop = React.createClass({
   },
 
   componentWillReceiveProps(nextProps, nextContext) {
-    let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
-
     this.setState({
       open: nextProps.open,
-      muiTheme: newMuiTheme,
+      muiTheme: nextContext.muiTheme || this.state.muiTheme,
     });
   },
 
-  getStyles() {
-    let {targetOrigin} = this.props;
-    let horizontal = targetOrigin.horizontal.replace('middle', 'vertical');
-
-    return {
-      base: {
-        opacity: 0,
-        transform: 'scaleY(0)',
-        transformOrigin: `${horizontal} ${targetOrigin.vertical}`,
-        position: 'fixed',
-        zIndex: this.state.muiTheme.zIndex.popover,
-        transition: Transitions.easeOut('450ms', ['transform', 'opacity']),
-        maxHeight: '100%',
-      },
-
-    };
-  },
-
-  getOpenStyles() {
-    return {
-      base: {
-        opacity: 1,
-        transform: 'scaleY(1)',
-      },
-    };
-  },
-
   render() {
-    let {
+    const {
       className,
       style,
       zDepth,
     } = this.props;
 
-    let styles = this.getStyles();
-    let openStyles = {};
-    if (this.state.open)
-      openStyles = this.getOpenStyles();
+    const styles = getStyles(this.props, this.state);
 
     return (
       <Paper
-        style={this.mergeStyles(styles.base, style, openStyles.base)}
```

Comment:
Sounds good, I'll remove [WIP] from the title because I wanted us to discuss that point first before merging. Thanks!